Frontend/android-native/mobility-common/src/main/java/in/juspay/mobility/common/MobilityCommonBridge.java

```diff
@@ -869,15 +939,60 @@ public void locateOnMap(boolean goToCurrentLocation, final String lat, final Str
                     googleMap.moveCamera(CameraUpdateFactory.newLatLngZoom(position, zoomLevel));
                     googleMap.moveCamera(CameraUpdateFactory.zoomTo(googleMap.getCameraPosition().zoom + 2.0f));
                 }
-
-                googleMap.setOnCameraIdleListener(() -> {
-                    double lat1 = (googleMap.getCameraPosition().target.latitude);
-                    double lng = (googleMap.getCameraPosition().target.longitude);
-                    if (storeLocateOnMapCallBack != null) {
-                        String javascript = String.format("window.callUICallback('%s','%s','%s','%s');", storeLocateOnMapCallBack, "LatLon", lat1, lng);
-                        bridgeComponents.getJsCallback().addJsToWebView(javascript);
-                    }
+                if (editLocation && googleMap != null) {
+                    removeAllPolylines("");
+                    CircleRippleEffectOptions options = new CircleRippleEffectOptions();
+                    options.radius(editPickupThreshold);
+                    options.strokeWidth(4);
+                    options.fromStrokeColor("#fcc32c");
+                    options.fillColor(0x22fcc32c);
+                    updateRippleCircleWithId("edit_location_circle", options, position);
+                }
+                googleMap.setOnCameraMoveStartedListener(new GoogleMap.OnCameraMoveStartedListener() {
+                    @Override
+                    public void onCameraMoveStarted(int i) {
+                        final ImageView labelView = labelId.equals("") ? null : Objects.requireNonNull(bridgeComponents.getActivity()).findViewById(Integer.parseInt(labelId));
+                       if (labelView != null)
+                           labelView.setVisibility(View.INVISIBLE);
+                        }
                 });
+                if (googleMap != null) {
+                    googleMap.setOnCameraIdleListener(() -> {
+                        double latitude = googleMap.getCameraPosition().target.latitude;
+                        double longitude = googleMap.getCameraPosition().target.longitude;
+                        LatLng source = new LatLng(Double.parseDouble(lat), Double.parseDouble(lon));
+                        double distance = SphericalUtil.computeDistanceBetween(googleMap.getCameraPosition().target, source);
+                        String key = "LatLon";
+                        if (editLocation) {
+                            removeMarker("ny_ic_src_marker");
+                            removeMarker("ny_ic_dest_marker");
+                            if (distance >= editPickupThreshold) {
+                                CircleRippleEffectOptions config = new CircleRippleEffectOptions();
+                                config.fromStrokeColor("#e55454");
```

Comment:
can we fetch this color from config(purescript). 
and also create a function which would return CircleRippleEffectOptions

Reply:
stroke color changes when you move out of circle, its not fixed or default

Reply:
we can send variables for both.